app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java

```diff
@@ -164,40 +171,85 @@ public AmazonS3Plugin(PluginWrapper wrapper) {
         }
 
         /*
-         * - Exception thrown here is handled by the caller.
+         * - Throws exception on upload failure.
+         * - Returns signed url of the created file on success.
          */
-        boolean uploadFileFromBody(AmazonS3 connection,
+        String uploadFileFromBody(AmazonS3 connection,
                                    String bucketName,
                                    String path,
-                                   String body)
-                                   throws InterruptedException {
-            InputStream inputStream = new ByteArrayInputStream(body.getBytes());
+                                   String body,
+                                   Boolean usingFilepicker,
+                                   int durationInMillis)
+                throws InterruptedException, AppsmithPluginException {
+
+            byte[] payload = null;
+            if(usingFilepicker == true) {
```

Comment:
The following ensures that the null value of usingFilepicker also gets caught.
```suggestion
            if(Boolean.TRUE.equals(usingFilepicker)) {
```